chapter_6/2_dictionary_learning.tex

```diff
@@ -0,0 +1,428 @@
+
+\section{Dictionary learning}
+\subsection{Introduction}
+Sparse dictionary learning is a representation learning method that aims at finding a sparse representation of the input data in the form of a linear combination of basic elements. These elements form the dictionary. 
+
+The problem of sparse coding consists of unsupervised learning of the dictionary and the coefficient matrices. Thus, given only unlabeled data, we aim to learn the set of dictionary atoms or
+basis functions that provide a good fit to the observed data. Sparse coding is important in a variety of domains. Sparse coding of natural images has yielded dictionary atoms which resemble the receptive fields of neurons in the visual cortex. Other important applications include compressed sensing and signal recovery.
+
+\subsection{Problem formulation}
+Dictionary learning has a particularly simple setup as follows. Given $n$ i.i.d. samples $y^i\in \mathbb{R}^d$, $i\in [n]$ from the generative model
+\begin{align}
+    y^i = A^* x^{i*}, \ i\in [n], \label{eq:gen-model}
+\end{align}
+the goal is to recover $A^*\in \mathbb{R}^{d\times r}$, the dictionary, and $s$-sparse vectors $x^{i*}\in \mathbb{R}^r$.  In many applications, the dictionary is often overcomplete, that is, $r \geq d$. Note that most optimization formulations of dictionary learning result in non-convex problems that have (often infinitely) many optimal solutions. This makes it difficult to analyze its optimization and generalization properties. 
+However, fixing one of $A$ and $x$, the problem reduces to linear regression involving response variable $y$ and covariates being either $A$ or $X$, for which various classical optimization methods for least-square problems can be used. This leads to various alternating minimization heuristics, which have been proven effective in practice. Meanwhile, theoretical guarantees for alternating minimization - such as guaranteed convergence to a global optimum and local rate of convergence - have also been developed.
+
+
+\section{Alternating minimization: algorithms and theoretical guarantees}
```

Comment:
I don't think you need to make this a new section. It's still part of dictionary learning.

Reply:
Sure.